Gems/Atom/RPI/Code/Source/RPI.Reflect/Material/LuaMaterialFunctor.cpp

```diff
@@ -68,6 +81,30 @@ namespace AZ
             LuaMaterialFunctorAPI::EditorContext::Reflect(behaviorContext);
         }
 
+        AZ::BehaviorContext* LuaMaterialBehaviorContext::GetBehaviorContext()
+        {
+            return m_sriptBehaviorContext.get();
+        }
+
+        void LuaMaterialFunctor::Reflect(ReflectContext* context)
+        {
+            if (auto* serializeContext = azrtti_cast<SerializeContext*>(context))
+            {
+                serializeContext->Class<LuaMaterialFunctor, RPI::MaterialFunctor>()
+                    ->Version(1)
+                    ->Field("scriptAsset", &LuaMaterialFunctor::m_scriptAsset)
+                    ->Field("materialNameContext", &LuaMaterialFunctor::m_materialNameContext)
+                    ;
+            }
+        }
+
+        LuaMaterialFunctor::LuaMaterialFunctor()
+        {
+            m_scriptContext = AZStd::make_unique<AZ::ScriptContext>(AZ_CRC_CE("MaterialFunctor"), &AZ::AllocatorInstance<AZ::SystemAllocator>::Get());
+            AZ::BehaviorContext* bc = LuaMaterialBehaviorContext::GetInstance()->GetBehaviorContext();
+            m_scriptContext->BindTo(bc);
```

Comment:
There is no need to even create a BehaviorContext or ScriptContext at all.
The global BehaviorContext would register the MaterialFunctor with it when the `Atom.RPI.Reflect` gem is activated

Reply:
It appears to be needed. Maybe there is a reason that we wanted to enclose some behaviors which are only accessible by material's lua scripts? 
Also, I have tried to use global BehaviorContext with adding those material functor related APIs and there were some issue about it.

Reply:
Well I am hitting the max allocator issue in my child allocator Jenkins AR run due to creating multiple ScriptContext (https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-16968/2/pipeline/843).
So I am looking into this issue now.